### PR TITLE
Add some avatar customization options in ProfileViewConfiguration

### LINKED ISSUE #245: Add an option to `ProfileViewConfiguration` for Avatar's borderWidth
Our avatar has 1pt border by default. I think this should be optional. Let's give others chance to set the `borderWidth` to whatever they want.

### PR DESCRIPTION
**Pull request by pinarol**
Closes #245 

### Description

I added following properties to `AvatarConfiguration`:

- cornerRadiusCalculator: A closure that calculates the corner radius of avatar based on its length.
- borderWidth
- borderColor
- avatarLength

I also removed some logic from DefaultAvatarProvider that waits until the avatar download ends to set the avatar border. I added that logic but I think it wasn't much necessary. I adjusted the placeholder implementation accordingly to hide the border during placeholder state. 

### Testing Steps

I added snapshot test but I also added a "Customize Avatar" switch to Demo App > Profile View controller. 
 Demo App> Profile Cards can also be checked to see if there are regressions.

### CODE REVIEW

**Comment by pinarol** [comment]
I was curious how/why snapshot of empty views has changed:

Screenshot 2024-06-05 at 15 47 44

The diff is on the border and hardly recognizable, I can't say I make much sense of it 🤔 

Screenshot 2024-06-05 at 15 48 02

File: Sources/GravatarUI/ProfileView/BaseProfileView.swift [hunk 1]
```diff
@@ -470,26 +543,17 @@ class DefaultAvatarProvider: AvatarProviding {
...
     func setImage(with source: URL?, placeholder: UIImage?, options: [ImageSettingOption]?) async throws {
-        do {
-            let _ = try await avatarImageView.gravatar.setImage(with: source, placeholder: placeholder, options: options)
-            if !skipStyling {
-                avatarImageView.layer.borderColor = paletteType.palette.avatar.border.cgColor
-                avatarImageView.layer.borderWidth = avatarBorderWidth
-            }
-        } catch {
-            if !skipStyling {
-                avatarImageView.layer.borderColor = UIColor.clear.cgColor
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/BaseProfileView.swift:489]: I removed this logic. I added this at the time but I think it turned out to be not necessary.

File: Sources/GravatarUI/ProfileView/Placeholder/PlaceholderDisplayers/BackgroundColorPlaceholderDisplayer.swift [hunk 2]
```diff
@@ -44,3 +44,19 @@ class BackgroundColorPlaceholderDisplayer<T: UIView>: PlaceholderDisplaying {
...
+    override func showPlaceholder() {
+        super.showPlaceholder()
+        originalBorderWidth = baseView.layer.borderWidth
+        baseView.layer.borderWidth = 0
+    }
+
+    override func hidePlaceholder() {
+        super.hidePlaceholder()
+        baseView.layer.borderWidth = originalBorderWidth
+    }
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/Placeholder/PlaceholderDisplayers/BackgroundColorPlaceholderDisplayer.swift:61]: Now that we have borders on the avatar from the start, we should hide them during the placeholder state.

**Review by etoledom — APPROVED** [review]
Looks pretty good! 🙏

**Comment by etoledom** [comment]
I like how the loading placeholders automatically adapt to the new shapes ✨

**Pull request merged**